File: Models/Contato.cs
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace AgendaAPI.Models{
    [Table("Contato")]
    public class Contato{
        [Key]
        [Column("ID")]
        public long Id { get; set; }

        [Required]
        [MaxLength(100)]
        [Column("NOME")]
        public string Nome { get; set; }

        [Column("IDADE")]
        public short Idade { get; set; }

        public List<Telefone> Telefones { get; set; } = new();
    }
}

File: src/types/Contato.ts
export interface Telefone {
  id?: number; 
  numero: string;
}

export interface Contato {
  id?: number; 
  nome: string;
  idade: number;
  telefones: Telefone[]; 
}

export interface ContatoSearch {
  nome?: string;
  numeroTelefone?: string;
}